### PR TITLE
Fix incorrect estimated time until play

### LINKED ISSUE #46: Incorrect estimated time until play
The estimated time until play is being calculated incorrectly - it should be the sum of the all the non-elapsed song durations before.
For instance, if song A is currently being played and is at `4:00 / 30:00` and song B is next and is `2:00` minutes long, then song C's estimated time until play should be `30:00 - 4:00 + 2:00 = 28:00`.

### Example

image

### PR DESCRIPTION
**Pull request by aquelemiguel**
* Closes #46

### CODE REVIEW

File: src/commands/play.rs [hunk 1]
```diff
@@ -102,6 +118,42 @@ pub async fn execute_play(
...
+            .expect("Could not get playing track info")
+            .position;
+
+        let top_track_duration = top_track
+            .metadata()
+            .duration
+            .expect("Could not fetch duration of top track");
+
+        let mut estimated_time = match flag {
+            PlayFlag::DEFAULT => queue[1..queue.len() - 1]
```
Comment (joao-conde) [inline src/commands/play.rs:137]: Can just do `queue[1..]`

File: src/commands/play.rs [hunk 2]
```diff
@@ -102,6 +118,42 @@ pub async fn execute_play(
...
+                    acc + x
+                        .metadata()
+                        .duration
+                        .expect("Could not fetch duration of track")
+                }),
+            PlayFlag::PLAYTOP => Duration::ZERO,
+        };
+
+        // Add the remaining top track
+        estimated_time += top_track_duration - top_track_position;
```
Comment (joao-conde) [inline src/commands/play.rs:149]: So `top_track_position` is not like a position index (0 1 2 or 3) right? I think its something else and perhaps rename it.

File: src/commands/play.rs [hunk 3]
```diff
@@ -102,6 +118,42 @@ pub async fn execute_play(
     Ok(())
 }
 
+async fn calculate_time_until_play(queue: &[TrackHandle], flag: &PlayFlag) -> Option<Duration> {
```
Comment (joao-conde) [inline src/commands/play.rs:121]: Very nice function, will become even better when we add the error propagation mechanisms and just unwrap values with `?`

**Review by joao-conde — CHANGES_REQUESTED** [review]

File: src/commands/play.rs (hunk 1 above)
Comment (aquelemiguel) [reply src/commands/play.rs:137]: Nope, I don't want the last track, as it is the last inserted track.

File: src/commands/play.rs (hunk 3 above)
Comment (aquelemiguel) [reply src/commands/play.rs:121]: 100% 🚀

File: src/commands/play.rs (hunk 2 above)
Comment (aquelemiguel) [reply src/commands/play.rs:149]: Good point. In this context, position is the elapsed time for a track (that's what Songbird calls it), but it does make you think it is a index. I'll rename it to something more clear.

File: src/commands/play.rs (hunk 1 above)
Comment (joao-conde) [reply src/commands/play.rs:137]: :donkey: me

**Review by joao-conde — APPROVED** [review]

**Pull request merged**